src/v037/server.rs

```diff
@@ -125,31 +199,77 @@ where
+        let conn = Connection {
+            consensus: self.consensus.clone(),
+            mempool: self.mempool.clone(),
+            info: self.info.clone(),
+            snapshot: self.snapshot.clone(),
+        };
+        //tokio::spawn(async move { conn.run(socket).await.unwrap() }.instrument(span));
+        tokio::spawn(async move { conn.run(socket).await.unwrap() });
+    }
+
+    pub async fn start(self) -> Result<(), BoxError> {
+        tracing::info!("starting ABCI server");
+
+        let tcp_addr = self
+            .tcp_listener
+            .as_ref()
+            .map(TcpListener::local_addr)
+            .transpose()
+            .map_err(|e| {
+                warn!(err.msg = %e, err.cause_chain = ?e, "failed getting tcp local addr");
+                e
+            })
+            .ok()
+            .flatten();
+        let uds_addr = self
+            .uds_listener
+            .as_ref()
+            .map(UnixListener::local_addr)
+            .transpose()
+            .map_err(|e| {
+                warn!(err.msg = %e, err.cause_chain = ?e, "failed getting uds local addr");
+                e
+            })
+            .ok()
+            .flatten();
+
+        tracing::info!(
+            addr.tcp = tcp_addr.map(debug),
+            addr.uds = uds_addr.map(debug),
+            "listening on local addresses"
+        );
 
         loop {
-            match listener.accept().await {
-                Ok((socket, _addr)) => {
-                    // set parent: None for the connection span, as it should
-                    // exist independently of the listener's spans.
-                    //let span = tracing::span!(parent: None, Level::ERROR, "abci", ?addr);
-                    let conn = Connection {
-                        consensus: self.consensus.clone(),
-                        mempool: self.mempool.clone(),
-                        info: self.info.clone(),
-                        snapshot: self.snapshot.clone(),
-                    };
-                    //tokio::spawn(async move { conn.run(socket).await.unwrap() }.instrument(span));
-                    tokio::spawn(async move { conn.run(socket).await.unwrap() });
+            select!(
+                tcp = async { self.tcp_listener.as_ref().unwrap().accept().await }, if self.tcp_listener.is_some() => {
+                    match tcp {
+                        Ok((socket, _addr)) => {
+                            self.handle_new_connection(socket);
+                        }
+                        Err(e) => {
+                            tracing::warn!({ %e }, "error accepting new tcp connection");
+                        }
+                    }
                 }
-                Err(e) => {
-                    tracing::warn!({ %e }, "error accepting new tcp connection");
+
+                uds = async { self.uds_listener.as_ref().unwrap().accept().await }, if self.uds_listener.is_some() => {
```

Comment:
I don't think we want to support two listeners, instead it would be simpler to have the consumer specify what socket they expect connections on, and have a simple `poll_accept`.

```rust

let listener = ...;
loop {
    match listener.accept().await {
        Ok((socket, _addr)) => {
            let conn = Connection {
                consensus: self.consensus.clone(),
                mempool: self.mempool.clone(),
                info: self.info.clone(),
                snapshot: self.snapshot.clone(),
            };
        tokio::spawn(async move { conn.run(socket).await.unwrap() });
    }
}
```

loop {
    match listener.accept() {